### DISCUSSION TITLE
HTTP server using socket.io with uwebsockets

### DISCUSSION THREAD

**Discussion by Backhie** [socketio/socket.io#5056] [Q&A] [answered]
Hi, I'm new to socket.io but I was already able to do some small projects using express + socket.io.
I'm trying to learn how to use uwebsockets with socket.io, the issue I'm facing is no easy way to use express in this setting.

At the moment after few hours of experimenting I ended up with using two ports - one for express http server and onether one for uwebsockets - socket.io communication.
It seems much more cumbersome than using plain socket.io with express. 
I'm creating new IO instance just for serving the socket.io.js for which I can't find any other solution.
I would really appreciate any guidance to push me in the right direction, is it the best way to achieve what i wanted?
Here's my backend code: 

```
import { App } from "uWebSockets.js";
import { Server } from "socket.io";
import express from "express";
import http from "http";

const port = 8000;
const ePort = 3000;
const eApp = express();
const server = http.createServer(eApp);
const app = App();

const io = new Server({
    cors: {
        origin: 'http://localhost:3000'
    }
});
const io2 = new Server(server);
eApp.use(express.static("public"));


io.attachApp(app);
io.on('connection', (socket) => {
    console.log(`New connection is player`);
    socket.emit('pStart', 1, 1);



    socket.on('disconnect', () => {
        console.log(`Player has disconnected`);

    });
});


eApp.get("/", (req, res) => {
    res.render("index.ejs");
});

app.listen(port, () => {
    console.log(`using port ${port}`)
})
server.listen(ePort, () => {
    console.log(`using port ${port}`)
})
```

and code from game.js to connect to uwebsockets - socket.io:
`const socket = io.connect(':8000');`

**Comment by darrachequesne** [comment, marked as answer]
Hi! I don't think this is possible. Express is meant to be used with a Node.js HTTP server, not with uwebsockets.

**Reply by Backhie** [reply]
> Hi! I don't think this is possible. Express is meant to be used with a Node.js HTTP server, not with uwebsockets.

Thank you so much for clarifying, I will keep making this project as it is for now, later I'll probably move to just uwebsockets + socket.io
